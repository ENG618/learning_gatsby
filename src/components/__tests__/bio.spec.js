import React from "react"
import { render } from "react-testing-library"
import { StaticQuery } from "gatsby" // mocked

import Bio from "../bio"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      avatar: {
        childImageSharp: {
          fixed: {
            width: 50,
            height: 50,
            src: `pretend-i-am-a-base64-encoded-image`,
            srcSet: `asdfasdf`,
          },
        },
      },
      site: {
        siteMetadata: {
          author: `me`,
          social: {
            twitter: `twitter_handle`,
          },
        },
      },
    })
  )
})

describe("Bio", () => {
  it("renders correctly", () => {
    const tree = render(<Bio />)
    expect(tree).toMatchSnapshot()
  })
})
