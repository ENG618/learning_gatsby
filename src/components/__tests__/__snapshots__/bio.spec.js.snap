// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; margin-bottom: 4.375rem;"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden; display: inline-block; width: 50px; height: 50px; margin-right: 0.875rem; margin-bottom: 0px; min-width: 50px; border-radius: 100%;"
        >
          <picture>
            <img
              alt="me"
              height="50"
              src="pretend-i-am-a-base64-encoded-image"
              srcset="asdfasdf"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s; border-radius: 50%;"
              width="50"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img width="50" height="50" srcset="asdfasdf" src="pretend-i-am-a-base64-encoded-image" alt="me" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
        <p>
          Written by 
          <strong>
            me
          </strong>
           who lives and works in San Francisco building useful things.
           
          <a
            href="https://twitter.com/twitter_handle"
          >
            You should follow him on Twitter
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; margin-bottom: 4.375rem;"
    >
      <div
        class=" gatsby-image-wrapper"
        style="position: relative; overflow: hidden; display: inline-block; width: 50px; height: 50px; margin-right: 0.875rem; margin-bottom: 0px; min-width: 50px; border-radius: 100%;"
      >
        <picture>
          <img
            alt="me"
            height="50"
            src="pretend-i-am-a-base64-encoded-image"
            srcset="asdfasdf"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s; border-radius: 50%;"
            width="50"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;img width="50" height="50" srcset="asdfasdf" src="pretend-i-am-a-base64-encoded-image" alt="me" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <p>
        Written by 
        <strong>
          me
        </strong>
         who lives and works in San Francisco building useful things.
         
        <a
          href="https://twitter.com/twitter_handle"
        >
          You should follow him on Twitter
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
